vortex-layout/src/layouts/zoned/reader.rs

```diff
@@ -136,18 +139,16 @@ impl ZonedReader {
             .clone()
     }
 
-    /// Return the zone range covered by a row range.
-    pub(crate) fn zone_range(&self, row_range: &Range<u64>) -> Range<usize> {
-        let zone_start = usize::try_from(row_range.start / self.layout.zone_len as u64)
-            .vortex_expect("Invalid zone start");
-        let zone_end = usize::try_from(row_range.end.div_ceil(self.layout.zone_len as u64))
-            .vortex_expect("Invalid zone end");
```

Comment:
we now defer doing the u64 -> usize truncation until we're actually building the Mask, which I think is slightly more correct but unlikely to come up in practice. Reduces a lot of usize::try_from at least 🤷